### PR TITLE
[skip changelog]Fix issue in ResourceManager and nopfsPlugin about repo path

### PR DESCRIPTION
**Pull request by fengzie**
For ResourceManager and nopfsPlugin, we need pass the Repo path for use, instead of getting from config.PathRoot(), which might not be the right one.

This will also fixes issue https://github.com/ipfs/kubo/issues/10407 .

### CODE REVIEW

**Comment by lidel** [comment]
@fengzie just to be sure this is triage properly, mind elaborating what you mean by "we need pass the Repo path for use, instead of getting from config.PathRoot(), which might not be the right one."? 

What would be example of state or use case where `config.PathRoot()` is "not be the right one"?

File: repo/repo.go [hunk 1]
```diff
@@ -23,6 +23,9 @@ type Repo interface {
 	// to the returned config are not automatically persisted.
 	Config() (*config.Config, error)
 
+	// Path is the repo file-system path
+	Path() string
```
Comment (lidel) [inline repo/repo.go:27]: What about in-memory or remote / virtual repos?

**Comment by fengzie** [comment]
> @fengzie just to be sure this is triage properly, mind elaborating what you mean by "we need pass the Repo path for use, instead of getting from config.PathRoot(), which might not be the right one."?
> 
> What would be example of state or use case where `config.PathRoot()` is "not be the right one"?

Hi @lidel , for `config.PathRoot()`, it gets the value from environment variable firstly, if empty it uses the default DefaultPathRoot. However, when creating an IPFS instance, user can specify the repo path instead of the default one.

If user create multiple instances through the SDK, then there might be issue.

 https://github.com/ipfs/kubo/blob/ce23fc7cccf1bea7ba0e93900b3654d2b879cad9/config/config.go#L57

File: repo/repo.go (hunk 1 above)
Comment (fengzie) [reply repo/repo.go:27]: Currently only FSRepo is provided. For the other kind of repo implementations, there should be design to be compatible with resourcemanager or nopfs.
https://github.com/ipfs/kubo/blob/ce23fc7cccf1bea7ba0e93900b3654d2b879cad9/repo/fsrepo/fsrepo.go#L366

**Review by gammazero — APPROVED** [review]

**Pull request merged**